core/services/gateway/connector/connector.go

```diff
@@ -46,7 +47,7 @@ type GatewayConnectorHandler interface {
 type gatewayConnector struct {
 	services.StateMachine
 
-	config      *ConnectorConfig
+	config      config.GatewayConnectorConfig
```

Comment:
This shouldn't need to be changed; keep using the struct (obviously you need the struct)